core/src/main/kotlin/org/evomaster/core/problem/rest/OpenApiAccess.kt

```diff
@@ -62,17 +63,40 @@ object OpenApiAccess {
     }
 
     private fun readFromDisk(openApiUrl: String) : String {
+
+        // file scheme
         val fileScheme = "file:"
-        val path = if (openApiUrl.startsWith(fileScheme, true)) {
-            Paths.get(URI.create(openApiUrl))
-        } else {
-            Paths.get(openApiUrl)
+
+        // path variable
+        var path: Path? = null;
+
+        // checking the given URL
+        try {
+            path = when (openApiUrl.startsWith(fileScheme, true)) {
+
+                true -> Paths.get(URI.create(openApiUrl));
+                false -> Paths.get(openApiUrl);
+            }
         }
-        if (!Files.exists(path)) {
-            throw SutProblemException("Cannot find OpenAPI schema at file location: $openApiUrl")
+        // if the URL is not a valid one
+        catch (e: Exception) {
+            throw SutProblemException("The file path provided for the OpenAPI Schema $openApiUrl," +
+                    " is not a valid path");
         }
 
-        return path.toFile().readText()
+        // Path should not be null, either
+        // it is set or an exception is thrown
+        if (path == null) {
+            throw SutProblemException("Could not set up the path: $openApiUrl");
+        }
+        else {
```

Comment:
unnecessary `else` declaration, as the `then` throws an exception (ie, avoid identation here)

Reply:
Hi Andrea,

Thank you so much for your feedback. I have removed the else clause from there.